src/lib/lightning/clightning/clightningService.ts

```diff
@@ -197,6 +199,43 @@ class CLightningService implements LightningService {
     );
   }
 
+  async addListenerToNode(node: LightningNode): Promise<void> {
+    console.log(`addListenerToNode ${node.implementation}`);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async removeListener(node: LightningNode): Promise<void> {
+    if (this.channelsInterval !== null) {
+      clearInterval(this.channelsInterval);
+    }
+  }
+
+  async subscribeChannelEvents(
+    node: LightningNode,
+    callback: (event: PLN.LightningNodeChannelEvent) => void,
+  ): Promise<void> {
+    // check c-lightning channels every 30 seconds
+    this.channelsInterval = setInterval(() => {
+      this.checkChannels(node, callback);
+    }, 3 * 10000);
```

Comment:
This is sharing one timer for all nodes, so only the last node started will receive events. Keep in mind that there will only ever be one instance of this service for the whole app, even when there are multiple nodes or multiple networks. So any state in here needs to be separated for each node.